wrappers/python/indy/crypto.py

```diff
@@ -391,80 +391,113 @@ def transform_cb(arr_ptr: POINTER(c_uint8), arr_len: c_uint32):
-    If it is an AnonAMES it will deserialize, decrypt, and return the message with an empty sender_vk string.
+    Packs a message by encrypting the message and serializes it in a JWE-like format (Experimental)
 
-    :param wallet_handle: wallet handler (created by open_wallet).
-    :param ames_json: a json string serialized using either AuthAMES or AnonAMES
-    :param my_vk: the verkey to authcrypt with
-    :return: message: the unencrypted message
-             sender_vk: the sender's verkey if AuthAMES, else an empty string
+    Note to use DID keys with this function you can call indy_key_for_did to get key id (verkey)
+    for specific DID.
+
+    #Params
+    command_handle: command handle to map callback to user context.
+    wallet_handle: wallet handler (created by open_wallet)
+    message: the output of a pack message
+
+    #Returns -> See HIPE 0028 for details
+    (Authcrypt mode)
+
+    {
+        "message": <decrypted message>,
+        "recipient_verkey": <recipient verkey used to decrypt>,
+        "sender_verkey": <sender verkey used to encrypt>
+    }
+
+    (Anoncrypt mode)
+
+    {
+        "message": <decrypted message>,
+        "recipient_verkey": <recipient verkey used to decrypt>,
+    }
     """
 
     logger = logging.getLogger(__name__)
-    logger.debug("unpack_message: >>> wallet_handle: %r, ames_json: %r, my_vk: %r",
+    logger.debug("unpack_message: >>> wallet_handle: %r, jwe: %r",
                  wallet_handle,
-                 ames_json,
-                 my_vk)
+                 jwe)
+
+    def transform_cb(arr_ptr: POINTER(c_uint8), arr_len: c_uint32):
+        return bytes(arr_ptr[:arr_len]),
+        
 
     if not hasattr(unpack_message, "cb"):
-        logger.debug("unpack_message: Creating callback")
-        unpack_message.cb = create_cb(CFUNCTYPE(None, c_int32, c_int32, c_char_p))
+        logger.debug("pack_message: Creating callback")
+        unpack_message.cb = create_cb(CFUNCTYPE(None, c_int32, c_int32, POINTER(c_uint8), c_uint32), transform_cb)
 
     c_wallet_handle = c_int32(wallet_handle)
-    c_ames_json = c_char_p(ames_json.encode('utf-8'))
-    c_my_vk = c_char_p(my_vk.encode('utf-8'))
-
-    message, sender_vk = await do_call('indy_unpack_messasge',
-                                       c_wallet_handle,
-                                       c_ames_json,
-                                       c_my_vk,
-                                       unpack_message.cb)
+    c_jwe_len = c_uint32(len(jwe))
+    unpack_message_bytes = await do_call('indy_unpack_message',
+                                         c_wallet_handle,
+                                         jwe,
+                                         c_jwe_len,
+                                         unpack_message.cb)
+
+    try:
+        unpack_message_str = unpack_message_bytes.decode('utf-8', 'strict')
```

Comment:
why do you do `decode('utf-8', 'strict')` on result of `unpack_message` but don't on result of `pack_message`?

Reply:
pack_message does not assume the type of encoding that will be used by the consumer because it doesn't assume the transport layer used. For example, in an HTTP Request, a JSON object can be put directly into the body of the request. However, if the transport layer is a URL then it should be base64URL encoded. However, in unpack_message it can be assumed that the API consumer will be processing the JSON object returned. In this case, we are assuming that the language will process JSON based on a specific type (string) in which case we can handle the encoding.

For consistency sake, I believe it should be the same across all APIs, so we should update the rust, .NET (PR #1437) and Java wrapper. Additionally, my first pass on the nodejs wrapper will need some help too (PR incoming).

Reply:
I strongly suggest to keep the same type for input parameter `msg` of pack and result of unpack.

and for current implementation I suggest to use bytes for pack/unpack as its current approach across all wrappers